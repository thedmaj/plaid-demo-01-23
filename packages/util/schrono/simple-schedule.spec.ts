import * as dateMock from 'jest-date-mock'
import {DateTime} from 'luxon'

import {
  formatSimpleScheduleExpression,
  parseSimpleScheduleExpression,
  SIMPLE_SCHEDULE_REGEX,
} from './simple-schedule'

const now = DateTime.local().set({year: 2020, month: 4, day: 13})

beforeAll(() => {
  dateMock.advanceTo(now.toJSDate())
})

afterAll(() => {
  dateMock.clear()
})

const sampleInputs = [
  'monthly from jan 1 to feb 23',
  'daily from jan 1 to feb 23',
  'daily jan 1 to feb 23',
  'daily jan 1 - feb 23',
  'daily jan 1',
  'jan 3',
  // this returns Jan 1, 2020, which is not correct but
  // still acceptable
  'weekly from dec 31 2020 to jan 1, 2020',
]

test.each(sampleInputs)('SIMPLE_SCHEDULE_REGEX: %o', (input) => {
  expect(input.match(SIMPLE_SCHEDULE_REGEX)?.slice(1)).toMatchSnapshot()
})

test.each(sampleInputs)('parseSimpleScheduleExpression: %o', (input) => {
  const se = parseSimpleScheduleExpression(input)

  expect({
    ...se,
    reformatted: formatSimpleScheduleExpression(se),
  }).toMatchSnapshot()
})
