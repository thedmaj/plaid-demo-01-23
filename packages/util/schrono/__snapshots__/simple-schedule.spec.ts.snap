// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SIMPLE_SCHEDULE_REGEX: 'daily from jan 1 to feb 23' 1`] = `
[
  "daily",
  "jan 1",
  "feb 23",
]
`;

exports[`SIMPLE_SCHEDULE_REGEX: 'daily jan 1 - feb 23' 1`] = `
[
  "daily",
  "jan 1",
  "feb 23",
]
`;

exports[`SIMPLE_SCHEDULE_REGEX: 'daily jan 1 to feb 23' 1`] = `
[
  "daily",
  "jan 1",
  "feb 23",
]
`;

exports[`SIMPLE_SCHEDULE_REGEX: 'daily jan 1' 1`] = `undefined`;

exports[`SIMPLE_SCHEDULE_REGEX: 'jan 3' 1`] = `undefined`;

exports[`SIMPLE_SCHEDULE_REGEX: 'monthly from jan 1 to feb 23' 1`] = `
[
  "monthly",
  "jan 1",
  "feb 23",
]
`;

exports[`SIMPLE_SCHEDULE_REGEX: 'weekly from dec 31 2020 to jan 1, 2020' 1`] = `
[
  "weekly",
  "dec 31 2020",
  "jan 1, 2020",
]
`;

exports[`parseSimpleScheduleExpression: 'daily from jan 1 to feb 23' 1`] = `
{
  "frequency": "days",
  "interval": "2020-01-01/2020-02-23",
  "reformatted": "Daily from Jan 1, 2020 to Feb 23, 2020",
}
`;

exports[`parseSimpleScheduleExpression: 'daily jan 1 - feb 23' 1`] = `
{
  "frequency": "days",
  "interval": "2020-01-01/2020-02-23",
  "reformatted": "Daily from Jan 1, 2020 to Feb 23, 2020",
}
`;

exports[`parseSimpleScheduleExpression: 'daily jan 1 to feb 23' 1`] = `
{
  "frequency": "days",
  "interval": "2020-01-01/2020-02-23",
  "reformatted": "Daily from Jan 1, 2020 to Feb 23, 2020",
}
`;

exports[`parseSimpleScheduleExpression: 'daily jan 1' 1`] = `
{
  "date": "2020-01-01",
  "reformatted": "Jan 1, 2020",
}
`;

exports[`parseSimpleScheduleExpression: 'jan 3' 1`] = `
{
  "date": "2020-01-03",
  "reformatted": "Jan 3, 2020",
}
`;

exports[`parseSimpleScheduleExpression: 'monthly from jan 1 to feb 23' 1`] = `
{
  "frequency": "months",
  "interval": "2020-01-01/2020-02-23",
  "reformatted": "Monthly from Jan 1, 2020 to Feb 23, 2020",
}
`;

exports[`parseSimpleScheduleExpression: 'weekly from dec 31 2020 to jan 1, 2020' 1`] = `
{
  "date": "2020-01-01",
  "reformatted": "Jan 1, 2020",
}
`;
